content/en/platform/corda/5.0-beta/developing/getting-started/first-flow/code-java.md

```diff
@@ -61,172 +61,197 @@ import net.corda.v5.application.messaging.FlowMessaging;
+  public MyFirstFlow() {}
 
-        // Log the message as a proxy for performing some useful operation on it.
-        log.info("MFF: Message received from " + receivedMessage.sender + ":" + receivedMessage.message);
+  // When a flow is invoked its call() method is called.
+  // Call() methods must be marked as @Suspendable, this allows Corda to pause mid-execution to wait
+  // for a response from the other flows and services.
 
-        // Get our identity from the MemberLookup service.
-        MemberX500Name ourIdentity = memberLookup.myInfo().getName();
+  @Suspendable
+  @Override
+  public String call(ClientRequestBody requestBody) {
 
-        // Create a response to greet the sender
-        Message response = new Message(ourIdentity,
-                "Hello " + session.getCounterparty().getCommonName() + ", best wishes from " + ourIdentity.getCommonName());
+    // Follow what happens in the console or logs.
+    log.info("MFF: MyFirstFlow.call() called");
 
-        // Log the response to be sent.
-        log.info("MFF: response.message: " + response.message);
+    // Show the requestBody in the logs - this can be used to help establish the format for starting a flow on Corda.
+    log.info("MFF: requestBody: " + requestBody.getRequestBody());
 
-        // Send the response via the send method on the flow session
-        session.send(response);
-    }
+    // Deserialize the Json requestBody into the MyfirstFlowStartArgs class using the JsonSerialisation service.
+    MyFirstFlowStartArgs flowArgs = requestBody.getRequestBodyAs(jsonMarshallingService, MyFirstFlowStartArgs.class);
+
+    // Obtain the MemberX500Name of the counterparty.
+    MemberX500Name otherMember = flowArgs.getOtherMember();
+
+    // Get our identity from the MemberLookup service.
+    MemberX500Name ourIdentity = memberLookup.myInfo().getName();
+
+    // Create the message payload using the MessageClass we defined.
+    Message message = new Message(otherMember, "Hello from " + ourIdentity + ".");
+
+    // Log the message to be sent.
+    log.info("MFF: message.message: " + message.getMessage());
+
+    // Start a flow session with the otherMember using the FlowMessaging service.
+    // The otherMember's virtual node will run the corresponding MyFirstFlowResponder responder flow.
+    FlowSession session = flowMessaging.initiateFlow(otherMember);
+
+    // Send the Payload using the send method on the session to the MyFirstFlowResponder responder flow.
+    session.send(message);
+
+    // Receive a response from the responder flow.
+    Message response = session.receive(Message.class);
+
+    // The return value of a RPCStartableFlow must always be a String. This will be passed
+    // back as the REST RPC response when the status of the flow is queried on Corda.
+    return response.getMessage();
+  }
 }
+
 /*
-RequestBody for triggering the flow via REST:
+RequestBody for triggering the flow via http-rpc:
```

Comment:
See prev comment

Reply:
I've also corrected this in all the other places it occurs